src/nl/hannahsten/texifyidea/inspections/latex/codematurity/LatexPrimitiveStyleInspection.kt

```diff
@@ -59,40 +61,53 @@ class LatexPrimitiveStyleInspection : TexifyInspectionBase() {
         return descriptors
     }
 
-    private inner class InspectionFix : LocalQuickFix {
+    private inner class InspectionFix(val oldCommand: SmartPsiElementPointer<LatexCommands>) : LocalQuickFixAndIntentionActionOnPsiElement(oldCommand.element) {
 
         @Nls
         override fun getFamilyName(): String {
             return "Convert to LaTeX alternative"
         }
 
-        override fun applyFix(
-            project: Project,
-            descriptor: ProblemDescriptor
-        ) {
-            val element = descriptor.psiElement as? LatexCommands ?: return
+        override fun getText(): String {
+            return oldCommand.element
+                ?.let { "Convert ${it.name} to ${CommandMagic.stylePrimitiveReplacements[it.name]}" }
+                ?: familyName
+        }
 
-            // Find elements that go after the primitive.
-            val cmdIndex = CommandMagic.stylePrimitives.indexOf(element.name)
-            if (cmdIndex < 0) {
-                return
-            }
-            val content = element.parentOfType(LatexNoMathContent::class)
-            val next = content!!.nextSiblingIgnoreWhitespace()
-            val after = if (next == null) "" else next.text
-            val replacement =
-                String.format(CommandMagic.stylePrimitveReplacements[cmdIndex], after)
-            val document =
-                PsiDocumentManager.getInstance(project).getDocument(element.containingFile)
+        override fun invoke(project: Project, file: PsiFile, editor: Editor?, startElement: PsiElement, endElement: PsiElement) {
+            (startElement as? LatexCommands)?.let { command ->
```

Comment:
To reduce nesting in the code by 8 spaces, would be more readable if you do
`if ((startElement as? LatexCommands).requiredParameters.isNotEmpty()) return`